ckan/controllers/package.py

```diff
@@ -388,6 +388,11 @@ def read(self, id):
         except (NotFound, NotAuthorized):
             abort(404, _('Dataset not found'))
 
+        # if the user specified a package id, redirect to the package name
+        if data_dict['id'] == c.pkg_dict['id']:
+            h.redirect_to(controller='package', action='read',
+                          id=c.pkg_dict['name'])
```

Comment:
Our site uses the same values for id and name, so this change would cause a redirect loop for us.

Reply:
Good catch! I'll fix this.